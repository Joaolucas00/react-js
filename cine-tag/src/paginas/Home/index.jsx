import styled from "styled-components"


const ConteinerFlex = styled.section`
    display: flex;
    h1 { 
        font-family: 'Montserrat';
        font-weight: 400;
    }
`

const TituloEstilizado = styled.h1`

`

const ConteudosCards = styled.section`

`

const Home = () => {
    return(
        <ConteinerFlex>
            <TituloEstilizado>Um lugar para guardar seus vÃ­deos!</TituloEstilizado>
            <ConteudosCards>

            </ConteudosCards>
        </ConteinerFlex>
    )
}

export default Home