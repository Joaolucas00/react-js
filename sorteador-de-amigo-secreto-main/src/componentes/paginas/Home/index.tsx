import Formulario from "../../Formulario"


const Home = () => {
    return (
        <>
            <Formulario/>
        </>
    )
}

export default Home