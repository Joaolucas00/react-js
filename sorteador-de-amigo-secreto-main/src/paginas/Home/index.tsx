import styled from "styled-components"
import Formulario from "../../componentes/Formulario"
import Rodape from "../../componentes/Rodape"

const MainEstilizado = styled.main`
    display: flex;
    background-color: white;
    width: 100%;
    height: 100vh;
    border: 1px solid black;
    border-radius: 10% 10% 0 0;
    flex-direction: column;
    align-items: center;
    align-content: center;
    justify-content: flex-start;
    flex-wrap: nowrap;
`

const Titulo = styled.h1`
    color: #4B69FD;
    font-size: 3rem;
    text-align: center;

`

const Home = () => {
    return (
        <MainEstilizado>
            <Titulo>Vamos come√ßar</Titulo>
            <Formulario/>
            <Rodape/>
        </MainEstilizado>
    )
}

export default Home