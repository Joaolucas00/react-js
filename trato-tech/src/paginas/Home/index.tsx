import Header from '../../componentes/Header';
import styles from './Home.module.scss';
import imagem from '../../assets/inicial.png';
import { useNavigate } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { AppDispatch, RootState } from '../../store';
import Botao from '../../componentes/Botao';
import { useEffect } from 'react';
import { carregarCategorias } from '../../store/reducers/categorias';

const Home = () => {


    const dispatch = useDispatch<AppDispatch>()

    useEffect(() => {
        dispatch(carregarCategorias())
    }, [dispatch])

    const navigate = useNavigate();
    const categorias = useSelector((state: RootState) => state.categorias)
    return (
        <div>
            <Header 
                titulo='Classificados Tech'
                descricao='Compre diversos tipos de produtos no melhor site do Brasil!'
                imagem={imagem}
                className={styles.header}
            >
            <Botao dataTestid='home-botao-anunciar' onClick={() => navigate('/anuncie')}>Quero anunciar</Botao>
            </Header>
            <div className={styles.categorias}>
                <div className={styles['categorias-title']}>
                    <h1>Categorias</h1>
                </div>
                <div className={styles['categorias-container']}>
                {categorias.map((categoria, index) => (
                    <div data-testid='home-categorias' key={index} onClick={() => navigate(`/categoria/${categoria.id}`)}>
                    <img src={categoria.thumbnail} alt={categoria.nome} />
                    <h1>{categoria.nome}</h1>
                    </div>
          ))}
                </div>
            </div>
        </div>
    )
}

export default Home