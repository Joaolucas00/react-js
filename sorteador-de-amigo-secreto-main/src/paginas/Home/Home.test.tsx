import { render } from "@testing-library/react"
import { RecoilRoot } from "recoil"
import Home from "."

const mockNavigate = jest.fn()

jest.mock('react-router-dom', () => {
    const reactRouter = jest.requireActual('react-router-dom')
    return {
        ...reactRouter,
        useNavigate: () => mockNavigate
    }
})

describe('A pagina de configuracao', () => {

    test('Deve ser renderizada corretamente', () => {
        const { container } = render(<RecoilRoot><Home/></RecoilRoot>)
        
        expect(container).toMatchSnapshot()
    })

})