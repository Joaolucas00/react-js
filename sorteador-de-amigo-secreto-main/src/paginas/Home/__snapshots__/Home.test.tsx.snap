// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A pagina de configuracao Deve ser renderizada corretamente 1`] = `
<div>
  <main
    class="sc-hLQSwg gqzbyT"
  >
    <h1
      class="sc-eDLKkx moga"
    >
      Vamos come√ßar
    </h1>
    <form
      class="sc-beySPh fyosFO"
    >
      <input
        class="sc-guDLey ekvGFt"
        placeholder="Insira os nomes dos participantes"
        type="text"
        value=""
      />
      <button
        class="sc-dmyCSP cnfAxP"
        disabled=""
      >
        Adicionar
      </button>
      
    </form>
    <footer>
      <button
        disabled=""
      >
        Iniciar brincadeira!
      </button>
    </footer>
  </main>
</div>
`;
